src/components/Radios.tsx

```diff
@@ -0,0 +1,282 @@
+  /**
+   * Provide attributes and classes to each radio item label.
+   */
+  label?: LabelProps;
+  /**
+   * Provide hint to each checkbox item.
+   */
+  hint?: HintProps;
+  /**
+   * Divider text to separate radio items, for example the text "or".
+   */
+  divider?: string;
+  /**
+   * If present, radio will be checked.
+   */
+  checked?: boolean;
+  /**
+   * If present content provided will be revealed when the item is checked.
+   * Maps to govuk's conditional prop
+   */
+  childrenWhenChecked?: React.ReactNode;
+  /**
+   * If true, radio will be disabled.
+   */
+  disabled?: boolean;
+}
+
+export interface RadiosProps extends React.AriaAttributes, DataAttributes {
+  /**
+   * Classes to add to the radio container.
+   */
+  className?: string;
+  /**
+   * Name attribute for each radio item.
+   */
+  name: string;
+  /**
+   * Array of {@link RadioButton} objects.
+   */
+  items: RadioButton[];
+  /**
+   * Options for the wrapping {@link Fieldset} component (e.g. legend).
+   */
+  fieldset?: Omit<FieldsetProps, "children">;
+  /**
+   * Options for the {@link Hint} component (e.g. text).
+   */
+  hint?: HintProps;
+  /**
+   * Options for the {@link ErrorMessage} component (e.g. text).
+   */
+  errorMessage?: ErrorMessageProps;
+  /**
+   * Options for the form-group wrapper
+   */
+  formGroup?: Omit<FormGroupProps, "children">;
+  /**
+   * String to prefix id for each checkbox item if no id is specified on each item.
+   * If `idPrefix` is not passed, fallback to using the name attribute instead.
+   */
+  idPrefix?: string;
+
+  onChange?(value: string): void;
+}
+
+export const Radios = (props: RadiosProps): JSX.Element => {
+  const {
+    className,
+    name,
+    items,
+    fieldset,
+    hint: radiosHint,
+    errorMessage,
+    onChange
+  } = props;
+  let formGroup = props.formGroup;
+  const idPrefix = props.idPrefix ? props.idPrefix : name;
+  const describedBy = [];
+  if (fieldset && fieldset.describedBy) {
+    describedBy.push(fieldset.describedBy);
+  }
+  let hintComponent: React.ReactNode, errorMessageComponent: React.ReactNode;
+  let hasConditional = false;
+  for (const radio of items) {
+    if (radio.childrenWhenChecked) {
+      hasConditional = true;
+    }
+  }
+
+  if (radiosHint) {
+    const hintId = radiosHint.id ? radiosHint.id : `${idPrefix}-hint`;
+    describedBy.push(hintId);
+    hintComponent = <Hint id={hintId} {...radiosHint} />;
+  }
+
+  if (errorMessage) {
+    const errorId = errorMessage.id ? errorMessage.id : `${idPrefix}-error`;
+    describedBy.push(errorId);
+    errorMessageComponent = <ErrorMessage id={errorId} {...errorMessage} />;
+    if (formGroup) {
+      formGroup.error = true;
+    } else {
+      formGroup = { error: true };
+    }
+  }
+
+  const extraAttributes = Attributes.ariaAndData(props);
+  const radioClassNames = classNames("govuk-radios lbh-radios", className, {
+    "govuk-radios--conditional": hasConditional
+  });
+
+  const renderDivider = (
```

Comment:
This, `renderRadio`, and `renderItem` could be pull to the outer scope to make the component render a bit easier to follow.

Reply:
If I pull them to the outer scope I have to pass through several more arguments to `renderItem` and `renderRadio` - do you think this is worth it?

Reply:
Yes, I think so. It also helps document what they need. You can make it a little cleaner by converting to taking a `props` argument with a strict type.